File: app/src/main/java/com/example/app_tema1_fredpor/menu.java
package com.example.app_tema1_fredpor;

import android.content.ActivityNotFoundException;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.Handler;
import android.provider.AlarmClock;
import android.service.autofill.UserData;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TableLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import java.util.Calendar;

public class menu extends AppCompatActivity {

    private ProgressBar ProgressBar;
    private int progressStatus = 0;
    private Handler handler = new Handler();
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.menu_main);
        ImageView imagenPerfil = findViewById(R.id.imagenPerfil);
        TextView textoBienvenida = findViewById(R.id.textoBienvenida);

        //inicio ini = new inicio();
        //Usuario user = ini.getUsuario();


        String sexo = usuario.getUserSex();
        String sexo = "H";


        //a partir de aqui funciona
        switch (sexo){
            case "H":
                imagenPerfil.setImageResource(R.drawable.avatar_messi);
                break;
            case "M":
                imagenPerfil.setImageResource(R.drawable.mujer);
                break;
            case "O":
                imagenPerfil.setImageResource(R.drawable.otro);
                break;
        }
        //String userName = user.getUserName();
        //textoBienvenida.setText(userName.toUpperCase());





        ImageButton botonDados = findViewById(R.id.botonDados);
        ImageButton botonAlarma = findViewById(R.id.botonAlarma);
        ImageButton botonModdle = findViewById(R.id.botonModdle);
        ImageButton botonMaps = findViewById(R.id.botonMaps);
        ImageButton botonLLamada = findViewById(R.id.botonLLamada);
        ImageButton botonChistes = findViewById(R.id.botonChistes);
        TableLayout tableLayout = findViewById(R.id.tableLayout);
        ProgressBar = findViewById(R.id.ProgressBar);


        new Thread(new Runnable() {
            public void run() {
                while (progressStatus < 100) {
                    progressStatus += 1;
                    handler.post(new Runnable() {
                        public void run() {
                            ProgressBar.setProgress(progressStatus);
                        }
                    });
                    try {
                        Thread.sleep(50);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }

                tableLayout.post(new Runnable() {
                    @Override
                    public void run() {
                        ProgressBar.setVisibility(View.INVISIBLE);
                       tableLayout.setVisibility(View.VISIBLE);
                    }
                });
            }
        }).start();


        botonDados.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(menu.this, dados.class);
                startActivity(intent);
            }
        });
        botonLLamada.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(menu.this, LLamadaActivity.class);
                startActivity(intent);
            }
        });

        botonChistes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(menu.this, chistes.class);
                startActivity(intent);
            }
        });


        botonAlarma.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Calendar cal = Calendar.getInstance();
                int horaActual = cal.get(Calendar.HOUR_OF_DAY);
                int minActual = cal.get(Calendar.MINUTE);

                minActual = minActual + 2;
                if (minActual >= 60){
                    minActual = minActual - 60;
                    horaActual++;
                    if (horaActual >= 24){
                        horaActual = horaActual -24;
                    }
                }
                Intent intent = new Intent(AlarmClock.ACTION_SET_ALARM);
                intent.putExtra(AlarmClock.EXTRA_MESSAGE, "ALARMA DE PRUEBA DE 2 MIN");
                intent.putExtra(AlarmClock.EXTRA_DAYS, Calendar.SUNDAY);
                intent.putExtra(AlarmClock.EXTRA_HOUR, horaActual);
                intent.putExtra(AlarmClock.EXTRA_MINUTES, minActual);

                try {
                    startActivity(intent);
                }catch (ActivityNotFoundException e){
                    e.printStackTrace();
                    Log.d("Error ", "no se ha podido crear la alarma");
                }
            }
        });

        botonModdle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url =  "https://educacionadistancia.juntadeandalucia.es/centros/jaen/";
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                startActivity(intent);
            }
        });

        botonMaps.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                double latitud = 40.4165;
                double longitud = -3.70256;
                String label = "Madrid";

                Uri IntentUri = Uri.parse("geo: " + latitud + "," + longitud + "?q=" + latitud + "," + longitud + "(" + label + ")");
                Intent map = new Intent(Intent.ACTION_VIEW, IntentUri );
                map.setPackage("com.google.android.apps.maps");
                startActivity(map);
            }
        });
    }
}

File: app/src/main/java/com/example/app_tema1_fredpor/Menu.kt
package com.example.app_tema1_fredpor

import android.content.ActivityNotFoundException
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.Handler
import android.provider.AlarmClock
import android.util.Log
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class Menu : AppCompatActivity() {

    private lateinit var progressBar: ProgressBar
    private var progressStatus = 0
    private val handler = Handler()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.menu_main)

        val imagenPerfil = findViewById<ImageView>(R.id.imagenPerfil)
        val imagenFooter = findViewById<ImageView>(R.id.imagenFooter)
        val textoBienvenida = findViewById<TextView>(R.id.textoBienvenida)
        val textoFooter = findViewById<TextView>(R.id.textoFooter)

        val usuario = ListarUser.obtenerUsuario()
        val user = usuario.firstOrNull()
        val userName = user?.userName
        val userSex = user?.userSex


        when (userSex) {
            "H" -> {
                imagenPerfil.setImageResource(R.drawable.avatar_messi)
                imagenFooter.setImageResource(R.drawable.messi_store)
                textoFooter.text = "THE MESSI STORE"
            }
            "M" -> {
                imagenPerfil.setImageResource(R.drawable.mujer)
                imagenFooter.setImageResource(R.drawable.shein)
                textoFooter.text = "SHEIN"
            }
            "O" -> {
                imagenPerfil.setImageResource(R.drawable.otro)
                imagenFooter.setImageResource(R.drawable.orapma)
                textoFooter.text = "ORAPMA PELUQUEROS DIGAME"
            }
        }
        textoBienvenida.text = userName?.toUpperCase()

        val botonDados = findViewById<ImageButton>(R.id.botonDados)
        val botonAlarma = findViewById<ImageButton>(R.id.botonAlarma)
        val botonModdle = findViewById<ImageButton>(R.id.botonModdle)
        val botonMaps = findViewById<ImageButton>(R.id.botonMaps)
        val botonLlamada = findViewById<ImageButton>(R.id.botonLLamada)
        val botonChistes = findViewById<ImageButton>(R.id.botonChistes)
        val tableLayout = findViewById<TableLayout>(R.id.tableLayout)
        progressBar = findViewById(R.id.ProgressBar)

        Thread {
            while (progressStatus < 100) {
                progressStatus += 1
                handler.post {
                    progressBar.progress = progressStatus
                }
                try {
                    Thread.sleep(50)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }

            tableLayout.post {
                progressBar.visibility = View.INVISIBLE
                tableLayout.visibility = View.VISIBLE
            }
        }.start()

        botonDados.setOnClickListener {
            val intent = Intent(this@Menu, Dados::class.java)
            startActivity(intent)
        }

        botonLlamada.setOnClickListener {
            val intent = Intent(this@Menu, LLamadaActivity::class.java)
            startActivity(intent)
        }

        botonChistes.setOnClickListener {
            val intent = Intent(this@Menu, Chistes::class.java)
            startActivity(intent)
        }

        botonAlarma.setOnClickListener {
            val cal = Calendar.getInstance()
            var horaActual = cal[Calendar.HOUR_OF_DAY]
            var minActual = cal[Calendar.MINUTE]

            minActual += 2
            if (minActual >= 60) {
                minActual -= 60
                horaActual++
                if (horaActual >= 24) {
                    horaActual -= 24
                }
            }
            val intent = Intent(AlarmClock.ACTION_SET_ALARM)
            intent.putExtra(AlarmClock.EXTRA_MESSAGE, "ALARMA DE PRUEBA DE 2 MIN")
            intent.putExtra(AlarmClock.EXTRA_DAYS, Calendar.SUNDAY)
            intent.putExtra(AlarmClock.EXTRA_HOUR, horaActual)
            intent.putExtra(AlarmClock.EXTRA_MINUTES, minActual)

            try {
                startActivity(intent)
            } catch (e: ActivityNotFoundException) {
                e.printStackTrace()
                Log.d("Error ", "no se ha podido crear la alarma")
            }
        }

        botonModdle.setOnClickListener {
            val url = "https://educacionadistancia.juntadeandalucia.es/centros/jaen/"
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            startActivity(intent)
        }

        botonMaps.setOnClickListener {
            val latitud = 40.4165
            val longitud = -3.70256
            val label = "Madrid"

            val intentUri = Uri.parse("geo: $latitud,$longitud?q=$latitud,$longitud($label)")
            val map = Intent(Intent.ACTION_VIEW, intentUri)
            map.`package` = "com.google.android.apps.maps"
            startActivity(map)
        }
    }
}